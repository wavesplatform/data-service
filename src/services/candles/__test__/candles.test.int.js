const { createPgDriver } = require('../../../db');
const loadConfig = require('../../../loadConfig');
const options = loadConfig();
const create = require('../index');

const amountAsset = 'AnERqFRffNVrCbviXbDEdzrU6ipXCP5Y1PKpFdRnyQAy';
const priceAsset = 'WAVES';

describe('Candles', () => {
  const service = create({
    drivers: { pg: createPgDriver(options) },
    emitEvent: () => () => null,
  });

  describe('search all candles between 2017-04-03 00:00:00 and 2017-04-03 23:59:59', () => {
    it('should return all Candles correctly for each 1h', done => {
      service
        .search({
          amountAsset,
          priceAsset,
          params: {
            timeStart: new Date('2017-04-03T00:00:00.000Z'),
            timeEnd: new Date('2017-04-03T23:59:59.999Z'),
            interval: '1h',
          },
        })
        .run()
        .listen({
          onResolved: value => {
            expect(value).toMatchSnapshot();
            done();
          }
        });
    });

    it('should return all Candles correctly for each 1d', done => {
      service
        .search({
          amountAsset,
          priceAsset,
          params: {
            timeStart: new Date('2017-04-03T00:00:00.000Z'),
            timeEnd: new Date('2017-04-03T23:59:59.999Z'),
            interval: '1d',
          },
        })
        .run()
        .listen({
          onResolved: value => {
            expect(value).toMatchSnapshot();
            done();
          }
        });
    });
  });
});
