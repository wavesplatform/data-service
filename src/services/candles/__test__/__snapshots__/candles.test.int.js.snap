// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candles search all candles between 2017-04-12 00:00:00 and 2017-04-12 23:59:59 should return all Candles correctly for each 1d 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T00:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
  ],
}
`;

exports[`Candles search all candles between 2017-04-12 00:00:00 and 2017-04-12 23:59:59 should return all Candles correctly for each 1h 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T00:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T01:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T02:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T03:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T04:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T05:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T06:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T07:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T08:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T09:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T10:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T11:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T12:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T13:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T14:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T15:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T16:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T17:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T18:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T19:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T20:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T21:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T22:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": "2017-04-12T23:00:00.000Z",
        "txsCount": 0,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
  ],
}
`;
